### PR TITLE
Pass original price in filters to improve compatibility

### LINKED ISSUE #28888: PHP 8 throws a fatal error "Unsupported operand types: string * int" in wc_price() when the price is an empty string
**Prerequisites (mark completed items with an [x]):**
- [x] I have have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.
- [x] It took me hours to figure out the root cause, because it's due to a breaking change in PHP 8, of which I wasn't aware. 😄

**Describe the bug**
Calling `wc_price('')`, passing an empty string to it, throws a fatal error in PHP 8, due to a change in how that PHP version handles different types.

**Expected behavior**
Up until PHP 7.4, calling wc_price('') would product the following output:
`<span class="woocommerce-Price-amount amount"><bdi><span class="woocommerce-Price-currencySymbol">$</span>0.00</bdi></span>`

**Actual behavior**
In PHP 8, calling `wc_price('')` throws the following fatal error:
`Unsupported operand types: string * int`

Root cause
The root cause of the issue is the following code
```
// When $price is an empty string, PHP up to 7.4 returns false, wherease PHP 8 returns true
$negative = $price < 0;

// In PHP 8, $negative is true here, therefore operation $price * -1 is attempted. Due to the
// stricter type checking "string * int" throws a fatal error. 
// In PHP 7.4 and earlier, the same operation would not have been performed and, even if 
// it had been, it would have thrown a warning, rather than an error
$price = apply_filters( 'raw_woocommerce_price', floatval( $negative ? $price * -1 : $price ) );
```

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
1. Prepare a test environment with PHP 8. **This is very important, as the bug doesn't occur with previous versions.**
2. Go to WooCommerce > Products > Add Product
3. Add a product.
4. Enter a product title, just to be able to identify the product.
5. Don't enter any price, anywhere. Although prices should be entered, up until PHP 7.4 an empty price would simply make the product unavailable, without throwing an error.
6. Try to preview the product. This should throw the fatal error.
7. Enter a regular price on the product.
8. Try to preview the product. The product should now appear correctly.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Suggestions on a possible fix**
Personally, I think that an empty price should be "left alone", as it's not a valid number and it should not be formatted. However, since until now WooCommerce formats such price as if it were zero, function wc_price() could be changed as follows:
```
// Store the original price passed as an argument (see below why)
$original_price = $price;
// Ensure that the $price is a number. If function wc_price() is called with "garbage",
// that will just set the price to zero
$price = floatval($price);

$unformatted_price = $price;
$negative          = $price < 0;
// Add the original raw price, before the floatval() call, as an argument to raw_woocommerce_price filter. This can
// simplify debugging, in case a price unexpectedly becomes zero after the floatval()
$price             = apply_filters( 'raw_woocommerce_price', $negative ? $price * -1 : $price, $original_price );
// Add the original raw price, before the floatval() call, as an argument to formatted_woocommerce_price filter. This can
// simplify debugging, in case a price unexpectedly becomes zero after the floatval()
$price             = apply_filters( 'formatted_woocommerce_price', number_format( $price, $args['decimals'], $args['decimal_separator'], $args['thousand_separator'] ), $price, $args['decimals'], $args['decimal_separator'], $args['thousand_separator'],  $original_price);
```

**WordPress Environment**
I didn't take a full report, because it's not completely relevant, but here is a summary of the test environment.
```
PHP 8 (this the most important element)
WordPress 5.6
WooCommerce 4.8.x/4.9.x
Storefront 3.0
```

### PR DESCRIPTION
**Pull request by claudiosanches**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This improves #28900 by keeping the original price in a variable and passing it to all filters to help developers.

Closes #28888.

### How to test the changes in this Pull Request:

See #28900

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: includes/wc-formatting-functions.php [hunk 1]
```diff
@@ -574,13 +574,33 @@ function wc_price( $price, $args = array() ) {
...
-	$price             = apply_filters( 'raw_woocommerce_price', floatval( $negative ? $price * -1 : $price ) );
-	$price             = apply_filters( 'formatted_woocommerce_price', number_format( $price, $args['decimals'], $args['decimal_separator'], $args['thousand_separator'] ), $price, $args['decimals'], $args['decimal_separator'], $args['thousand_separator'] );
+
+	/**
+	 * Filter raw price.
+	 *
+	 * @param float        $raw_price      Raw price.
+	 * @param float|string $original_price Original price as float, or empty string. Since 5.0.0.
+	 */
+	$price = apply_filters( 'raw_woocommerce_price', floatval( $negative ? $price * -1 : $price ), $original_price );
```
Comment (roykho) [inline includes/wc-formatting-functions.php:591]: This extra `floatval()` is redundant because line 580 already converted.

**Review by roykho — CHANGES_REQUESTED** [review]
Just a small comment.

File: includes/wc-formatting-functions.php (hunk 1 above)
Comment (claudiosanches) [reply includes/wc-formatting-functions.php]: Good catch, thanks!

**Review by roykho — APPROVED** [review]
LGTM!

**Pull request merged**